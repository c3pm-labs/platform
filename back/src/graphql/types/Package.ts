import { objectType } from '@nexus/schema';

import { PackageService } from '../../services/PackageService';

export const Versions = objectType({
  name: 'Version',
  definition(t) {
    t.model.description();
    t.model.license();
    t.model.package();
    t.model.readme();
    t.model.publishedAt();
    t.model.version();
  },
});

export const Package = objectType({
  name: 'Package',
  definition(t) {
    t.model.author();
    t.model.name();
    t.model.versions();
    t.field('latest', {
      type: 'Version',
      resolve: async (pkg, _args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.getLatestVersion(pkg.name);
      },
    });
  },
});
