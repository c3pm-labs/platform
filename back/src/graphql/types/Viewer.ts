import { objectType } from '@nexus/schema';

export const Viewer = objectType({
  name: 'Viewer',
  definition(t) {
    t.model('User').id();
    t.model('User').email();
    t.model('User').username();
  },
});
