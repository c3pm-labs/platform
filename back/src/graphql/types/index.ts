export * from './User';
export * from './Mutation';
export * from './Query';
export * from './Package';
