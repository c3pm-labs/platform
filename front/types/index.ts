export interface Package {
  name: string;
  latest: Version;
  versions: Version[];
  author: User;
  downloads: number;
  tags: string[];
}

export interface Version {
  version: string;
  publishedAt: string;
  description: string;
  readme: string;
  package?: Package;
}

export interface User {
  id: string;
  username: string;
  email: string;
  description: string;
  packages: Package[];
}

export interface Viewer extends User {
  username: string;
  email: string;
}
