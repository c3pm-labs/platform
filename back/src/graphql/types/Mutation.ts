import { mutationType, stringArg } from '@nexus/schema';

import { AuthService } from '../../services/AuthService';
import { UpdateService } from '../../services/UpdateService';

import { Viewer } from './Viewer';
import { User } from './User';

export const Mutation = mutationType({
  definition(t) {
    t.field('register', {
      type: Viewer,
      args: {
        username: stringArg({ required: true }),
        email: stringArg({ required: true }),
        password: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.register(args);
      },
    });
    t.field('login', {
      type: Viewer,
      args: {
        login: stringArg({ required: true }),
        password: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.login(args);
      },
    });
    t.field('logout', {
      type: Viewer,
      resolve: (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.logout();
      },
    });
    t.field('update', {
      type: User,
      nullable: true,
      args: {
        id: stringArg(),
        username: stringArg(),
        email: stringArg(),
        description: stringArg(),
      },
      resolve: async (parent, args, ctx) => {
        const updateService = new UpdateService(ctx);
        return updateService.updateUser(args);
      },
    });
  },
});
