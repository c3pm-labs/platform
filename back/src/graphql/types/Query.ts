import { queryType, stringArg } from '@nexus/schema';

import { AuthService } from '../../services/AuthService';
import { UserService } from '../../services/UserService';
import { PackageService } from '../../services/PackageService';

import { Viewer } from './Viewer';
import { User } from './User';
import { Package, Versions } from './Package';

export const Query = queryType({
  definition(t) {
    t.field('viewer', {
      type: Viewer,
      resolve: async (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.viewer();
      },
    });
    t.field('user', {
      type: User,
      nullable: true,
      args: {
        username: stringArg(),
        id: stringArg(),
      },
      resolve: async (parent, args, ctx) => {
        const userService = new UserService(ctx);
        return userService.findOne(args);
      },
    });
    t.list.field('search', {
      type: Package,
      args: {
        keyword: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.search(args.keyword);
      },
    });
    t.field('package', {
      type: Package,
      args: {
        name: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.getPackage(args.name);
      },
    });
    t.field('version', {
      type: Versions,
      args: {
        packageName: stringArg({ required: true }),
        version: stringArg({ required: false }),
      },
      resolve: async (parent, args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.getVersionOrLatest(args.packageName, args.version);
      },
    });
  },
});
