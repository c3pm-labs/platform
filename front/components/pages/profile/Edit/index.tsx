import { Form, Formik } from 'formik';
import { Box, makeStyles } from '@material-ui/core';
import { Dispatch, SetStateAction } from 'react';

import TextInput from '../../../TextInput';
import Button from '../../../Button';
import { User } from '../../../../types';

const useStyles = makeStyles((theme) => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    marginTop: theme.spacing(3),
  },
  description: {
    display: 'flex',
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
    fontSize: 16,
    color: theme.palette.grey[600],
  },
  save: {
    display: 'flex',
    minWidth: 100,
  },
  cancel: {
    display: 'flex',
    minWidth: 100,
    marginLeft: theme.spacing(1),
  },
}));

export interface EditProps {
  setIsEdit: Dispatch<SetStateAction<boolean>>;
  user: User;
}

const Edit = ({ setIsEdit, user }: EditProps): JSX.Element => {
  const classes = useStyles();
  const onSubmit = null;

  return (
    <>
      <Formik
        initialValues={{
          username: user.username,
          email: user.email,
          description: user.description ? user.description : '',
        }}
        {...onSubmit}
      >
        <Form className={classes.form}>
          <TextInput
            name="username"
            label="username"
            placeholder={user.username}
            required
            fullWidth
            type="text"
          />
          <TextInput
            name="email"
            label="email"
            placeholder={user.email}
            required
            fullWidth
            type="email"
          />
          <TextInput
            name="description"
            label="description"
            placeholder={user.description}
            required={false}
            fullWidth
            type="text"
            multiline
            rows={4}
          />
        </Form>
      </Formik>
      <Box display="flex" justifyContent="center" flexDirection="row">
        <Button
          className={classes.save}
          color="primary"
          variant="contained"
          type="submit"
          onClick={(): void => setIsEdit(false)}
        >
          Save
        </Button>
        <Button
          className={classes.cancel}
          color="error"
          variant="contained"
          type="submit"
          onClick={(): void => setIsEdit(false)}
        >
          Cancel
        </Button>
      </Box>
    </>
  );
};

export default Edit;
