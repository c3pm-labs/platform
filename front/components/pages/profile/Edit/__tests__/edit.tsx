import { render } from '@testing-library/react';
import { User } from 'types';

import Edit from '../index';

test('Edit', () => {
  const user = {
    id: '1234',
    username: 'toto',
    email: 'toto@gmail.com',
    description: 'make jokes',
    packages: {
      name: '',
      latest: {
        version: '0.2',
        publishedAt: '10:04:1999',
        description: 'hot fix',
        readme: 'blablabla',
      },
      versions: [
        {
          version: '0.2',
          publishedAt: '10:04:1999',
          description: 'hot fix',
          readme: 'blablabla',
        },
        {
          version: '0.2',
          publishedAt: '10:04:1999',
          description: 'hot fix',
          readme: 'blablabla',
        },
      ],
      author: {
        id: '12345',
        username: 'Bob',
        email: 'bob.dylan@gmail.com',
        description: 'singer',
        packages: [],
      },
    },
  };

  const { container } = render(
    <Edit
      user={(user as unknown) as User}
      // eslint-disable-next-line no-console
      setIsEdit={() => console.log('toto')}
    />,
  );

  expect(container).toMatchSnapshot();
});
