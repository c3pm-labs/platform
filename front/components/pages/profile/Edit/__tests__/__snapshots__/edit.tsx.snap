// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit 1`] = `
<div>
  <form
    action="#"
    class="makeStyles-form-1"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink="true"
        for="username"
        id="username-label"
      >
        username
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-describedby="username-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="username"
          name="username"
          placeholder="toto"
          required=""
          type="text"
          value="toto"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              username
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-root-5 MuiFormHelperText-contained MuiFormHelperText-filled Mui-required MuiFormHelperText-marginDense"
        id="username-helper-text"
      >
        <span>
          ​
        </span>
      </p>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink="true"
        for="email"
        id="email-label"
      >
        email
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-describedby="email-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="email"
          name="email"
          placeholder="toto@gmail.com"
          required=""
          type="email"
          value="toto@gmail.com"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              email
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-root-5 MuiFormHelperText-contained MuiFormHelperText-filled Mui-required MuiFormHelperText-marginDense"
        id="email-helper-text"
      >
        <span>
          ​
        </span>
      </p>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink="true"
        for="description"
        id="description-label"
      >
        description
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <textarea
          aria-describedby="description-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="description"
          name="description"
          placeholder="make jokes"
          rows="4"
        >
          make jokes
        </textarea>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              description
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-root-5 MuiFormHelperText-contained MuiFormHelperText-filled MuiFormHelperText-marginDense"
        id="description-helper-text"
      >
        <span>
          ​
        </span>
      </p>
    </div>
  </form>
  <div
    class="MuiBox-root MuiBox-root-10"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-11 makeStyles-primary-12 makeStyles-primary-16 makeStyles-save-3 MuiButton-disableElevation"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Save
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-11 makeStyles-error-13 makeStyles-error-21 makeStyles-cancel-4 MuiButton-disableElevation"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Cancel
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
